cmd/terramate/cli/project.go

```diff
@@ -39,7 +39,7 @@ type project struct {
 }
 
 func (p project) gitcfg() *hcl.GitConfig {
-	return p.rootcfg.Terramate.RootConfig.Git
+	return p.rootcfg.Terramate.Config.Git
```

Comment:
since these parsed structure doens't necessarily represents a root config, but any config block on any part of the hierarchy, I renamed it. I think we can also rename the types to reflect that, but will do that in a separate PR because the changeset is quite big.

Reply:
what about the issue we were talking earlier today? we talked about failing if the user has `terramate.config` in a non-root directory.. but the `run.env` feature kinda requires that.

Reply:
`terramate.config` is not a root configuration. The root configurations are:

* terramate.required_version
* terramate.config.git

At least as documented today. If everything inside `terramate.config` is supposed to be root only then we need to come up with a different place to put the run/env configurations.

Reply:
Maybe we should have a project_config block to make it explicit that it has project wide configuations that can only be defined on the top level/root dir ? :thinking:

Reply:
we discussed the following options with the team:

* Only allow run/env on root and update specs (but if we want hierarchical in the future the problem comes back)
* Handle as project root terramate.required_version OR terramate.config.git instead of overall existence of terramate.config
* Move terramate.config.run.env to some other place not inside terramate.config and keep it as is
* Move `terramate.config` to something like terramate.project to make it more explicit that it is a project wide/root only config
* Make terramate.config a non-root config that is hierarchical + change project root detection

We ended up deciding for "Only allow run/env on root and update specs" for now. So `terramate.config.run.env` will be a root only configuration for now, no hierarchical/merge behavior.

Reply:
awesome.
:rocket: